apps/pay/app/api/auth/[...nextauth]/auth.ts

```diff
@@ -66,5 +66,15 @@ export const authOptions: NextAuthOptions = {
       session.accessToken = token.accessToken
       return session
     },
+    async signIn({ account }) {
+      if (!account?.access_token) {
+        return false
+      }
+      const res = await fetchUserData({ token: account.access_token })
+      if (res instanceof Error || !res.data.me?.username) {
+        return "/error?errorMessage=This account does not have a username. Please update your profile from the mobile app and update your username"
+      }
```

Comment:
I think there are 2 errors possible here. if the server doesn't respond for whatever reason, it will return an error. but the error message `This account does not have a username` will be wrong